### DISCUSSION TITLE
Pecent-encoded

### DISCUSSION THREAD

**Discussion by ramzh** [cross-seed/cross-seed#353] [Q&A] [answered]
When setting up the qBittorrentUrl using the docs format:
http://username:password@192.168.43.11:8081

an error is thrown:
`error: qBittorrent url must be percent-encoded`

Even encoding the URL string returns the same error

OS: OpenMediaVault Shaitain
Installation: Standalone Docker-Compose/Stack via Portrainer
Network: same network as qBittorrent

**Comment by mmgoodnow** [comment]
The entire URL must be percent encoded, including the username and password.

**Comment by ramzh** [comment]
Still getting the same error with the following:

http%3A%2F%2Fusername%3Apassword%40192.168.43.11%3A8081

**Reply by mmgoodnow** [reply]
Sorry, I wasn't clear. The parts that are URL syntax (`http://`, the `:` in between your username and password) still need to be unencoded. If I had to guess, the problem is the password if your username doesn't contain any special characters. 

`http://{{ username }}:{{ urlencoded password }}@host:port/path`

**Reply by ramzh** [reply]
I do still have special characters. Anyway I could handle it?

**Reply by mmgoodnow** [reply, marked as answer]
Then it's `http://{{ urlencoded username }}:{{ urlencoded password }}@host:port/path`

**Reply by ramzh** [reply]
Ok working fine! Thanks, just need to solve the indexer issue now!
